File: CyFile-Android/app/src/test/java/at/tugraz/tc/cyfile/crypto/PrefixCryptoServiceUnitTest.java
package at.tugraz.tc.cyfile.crypto;

import junit.framework.Assert;

import org.junit.Before;
import org.junit.Test;

import at.tugraz.tc.cyfile.BaseUnitTest;
import at.tugraz.tc.cyfile.crypto.impl.AESCryptoService;
import at.tugraz.tc.cyfile.crypto.impl.ApacheCodecBase64;
import at.tugraz.tc.cyfile.crypto.impl.DummyKeyVaultService;
import at.tugraz.tc.cyfile.crypto.impl.PrefixCryptoService;

import static org.mockito.Mockito.spy;

/**
 * Created by mmalte on 05.06.18.
 */

public class PrefixCryptoServiceUnitTest extends BaseUnitTest {

    private PrefixCryptoService cryptoService;

    @Before
    public void setup() {
        cryptoService = new PrefixCryptoService();
    }

    @Test
    public void testEncryptWithStringReturnsWithPrefix() {
        String test = "test";
        String actual = cryptoService.encrypt(test);
        Assert.assertTrue(actual.contains("test"));
        Assert.assertTrue(actual.contains("PLEASE DON'T READ"));
    }

    @Test
    public void testDecryptWithStringReturnsWithoutPrefix() {
        String test = "PLEASE DON'T READ test";
        String actual = cryptoService.decrypt(test);
        Assert.assertTrue(actual.contains("test"));
        Assert.assertFalse(actual.contains("PLEASE DON'T READ"));
    }

}

File: CyFile-Android/app/src/test/java/at/tugraz/tc/cyfile/crypto/PrefixCryptoServiceUnitTest.kt
package at.tugraz.tc.cyfile.crypto

import at.tugraz.tc.cyfile.BaseUnitTest
import at.tugraz.tc.cyfile.crypto.impl.PrefixCryptoService
import org.junit.Assert
import org.junit.Before
import org.junit.Test

/**
 * Created by mmalte on 05.06.18.
 */

class PrefixCryptoServiceUnitTest : BaseUnitTest() {

    private var cryptoService: PrefixCryptoService? = null

    @Before
    fun setup() {
        cryptoService = PrefixCryptoService()
    }

    @Test
    fun testEncryptWithStringReturnsWithPrefix() {
        val test = "test"
        val actual = cryptoService!!.encrypt(test)
        Assert.assertTrue(actual.contains("test"))
        Assert.assertTrue(actual.contains("PLEASE DON'T READ"))
    }

    @Test
    fun testDecryptWithStringReturnsWithoutPrefix() {
        val test = "PLEASE DON'T READ test"
        val actual = cryptoService!!.decrypt(test)
        Assert.assertTrue(actual.contains("test"))
        Assert.assertFalse(actual.contains("PLEASE DON'T READ"))
    }

}
